File: gems/decomplex/examples/c/structural-topology.c
typedef struct Worker { int ready_flag; } Worker;
void run(Worker *self, Items items) { prepare(self); if (ready(self)) { validate(self); } for (int item = 0; item < items.count; item++) { helper(self, item); } }
void prepare(Worker *self) {}
bool ready(Worker *self) { return true; }
void validate(Worker *self) {}
void helper(Worker *self, Item item) { item.use(); }

File: gems/decomplex/examples/python/structural-topology.py
class Worker:
    def run(self, items):
        self.prepare()
        if self.ready():
            self.validate()
        for item in items:
            self.helper(item)
    def prepare(self): pass
    def ready(self): return True
    def validate(self): pass
    def helper(self, item): return item
